test/Spec/BotPlutusInterface/Contract.hs

```diff
@@ -851,9 +870,27 @@ assertCommandHistory :: forall (w :: Type). MockContractState w -> [(Int, Text)]
 assertCommandHistory state =
   mapM_
     ( \(idx, expectedCmd) ->
-        (state ^? commandHistory . ix idx) @?= Just (Text.replace "\n" " " expectedCmd)
+        assertCommandEqual
+          ("command at index " ++ show idx ++ " was incorrect:")
+          (Text.replace "\n" " " expectedCmd)
+          (fromMaybe "" $ state ^? commandHistory . ix idx)
     )
 
+assertCommandEqual :: String -> Text -> Text -> Assertion
+assertCommandEqual err expected actual
+  | commandEqual expected actual = return ()
+  | otherwise = assertFailure $ err ++ "\nExpected:\n" ++ show expected ++ "\nGot:\n" ++ show actual
+
+commandEqual :: Text -> Text -> Bool
+commandEqual "" "" = True
+commandEqual "" _ = False
+commandEqual _ "" = False
+commandEqual expected actual = maybe False (on commandEqual dropToSpace postExp) mPostAct
+  where
+    (preExp, postExp) = Text.breakOn "?" expected
+    mPostAct = Text.stripPrefix preExp actual
+    dropToSpace = Text.dropWhile (/= ' ')
```

Comment:
This could likely be done better, didn't want to bring all of regex in though, any ideas?

Reply:
I can't come up with anything better right how either.
However, I think we could have the trimming of newlines on line 875 merged into this function

Reply:
Oh yeah, good point